include/toolchain/common.h

```diff
@@ -98,7 +98,8 @@
 /* Indicate that an array will be used for stack space. */
 
 #if !defined(_ASMLANGUAGE)
-  #define __stack __aligned(STACK_ALIGN)
+  /* don't use this anymore, use K_DECLARE_STACK instead. Remove for 1.11 */
```

Comment:
Anas, this doesn't work either, the preprocessor does not expand ZEPHYR_VERSION the way you expect:

/projects/zephyr2/include/toolchain/common.h:102:43: error: missing binary operator before token "("
 #if (ZEPHYR_VERSION_CODE <= ZEPHYR_VERSION(1,11,0))
                                           ^
/projects/zephyr2/include/toolchain/common.h:100:0: error: unterminated #if
 #if !defined(_ASMLANGUAGE)
 
Also, as noted below, what Leandro is proposing is not actually possible with the preprocessor.

I ask we leave this macro as-is and focus on the memory protection features in this patch set.